userspace/utilities/displayctl.cpp

```diff
@@ -95,53 +95,52 @@ int gfxmode_set_compositor(IOCallDisplayModeArgs mode)
 
     connection.write(&goodbye_message, sizeof(goodbye_message));
 
-    return PROCESS_SUCCESS;
+    return Result::SUCCESS;
 }
 
-int gfxmode_set_iocall(Stream *device, IOCallDisplayModeArgs mode)
+Result gfxmode_set_iocall(Stream *device, IOCallDisplayModeArgs mode)
 {
     if (stream_call(device, IOCALL_DISPLAY_SET_MODE, &mode) != SUCCESS)
     {
         handle_printf_error(device, "Ioctl to " FRAMEBUFFER_DEVICE_PATH " failed");
-        return PROCESS_FAILURE;
+        return Result::ERR_INAPPROPRIATE_CALL_FOR_DEVICE;
     }
 
-    return PROCESS_SUCCESS;
+    return Result::SUCCESS;
 }
 
-int gfxmode_set(String &mode_name)
+ArgParseResult gfxmode_set(String &mode_name)
```

Comment:
This feels wrong, the ArgParseResult abstraction is leaking out of the ArgParse scope

Reply:
Yep, will add it as enum class to **ArgParse**

Reply:
Turns out adding it into **ArgParse** gives conflicts with declaring **ArgParseOptionCallback** earlier

Reply:
I need to clarify,  the thing that's bothering me here is that ArgParseResult is supposed to be the results of an arg parse but here it's used to instruct ArgParse what it should do next.

Reply:
Yes, i wanted it to both but's kind of a hack.... **eval** needs to return the 3 modes (Continue, Done, Failure). Since each option somehow needs to tell what "kind" of option it is hacked this into there. Maybe we should add this to the **option** itself and just make these return a regular **Result**? So something like this:
```c++
    args.option_string('s', "set", "Set graphic mode.", ArgParseOptionResult::SHOULD_FINISH, [&](String &mode) {
        return gfxmode_set(mode); // This returns RESULT
    })
```

Reply:
We can return an Utils::Variant<Result, ArgParseResult> ?